### PR TITLE
[Refactor] Add FeatureMapProcessor to replace transform_inputs in heads

### PR DESCRIPTION
**Pull request by Ben-Louis**
## Modification

1. Introduced a new neck, `FeatureMapProcessor`, which is capable of selecting, concatenating, and rescaling feature maps generated by backbone layers.
2. Removed the `_transform_inputs` and `_get_in_channels` functions from heads.
3. Updated configs according to the above modifications. 

The test accuracy of several representative models affected by these changes has been verified:
- `cid_hrnet-w32_8xb20-140e_coco-512x512`
- `dekr_hrnet-w32_8xb10-140e_coco-512x512`
- `dekr_hrnet-w32_8xb10-300e_crowdpose-512x512`
- `ipr_res50_8xb64-210e_coco-256x256`
- `rtmpose-t_8xb256-420e_coco-256x192`
- `simcc_res50_8xb64-210e_coco-256x192`
- `td-hm_hrnet-w32_8xb64-210e_coco-256x192`
- `td-hm_pvt-s_8xb64-210e_coco-256x192`
- `td-hm_ViTPose-base-simple_8xb64-210e_coco-256x192`
- `td-reg_res50_8xb64-210e_coco-256x192`
- `td-reg_res50_rle-pretrained-8xb64-210e_coco-256x192`

## Checklist

**Before PR**:

- [ ] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit tests to ensure correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Comment by Tau-J** [comment]
I suggest using `rtmpose-s` or `rtmpose-m` by defaults.

**Comment by Tau-J** [comment]
`FeatureMapProcessor` is a newly introduced module, and I suggest adding an explanation of it to the [20-minute tutorial](https://mmpose.readthedocs.io/en/1.x/guide_to_framework.html#neck) to help new users understand and use it. There's a part

File: mmpose/models/necks/fmap_proc_neck.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from typing import List, Optional, Sequence, Tuple, Union
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torch import Tensor
+
+from mmpose.models.utils.ops import resize
```
Comment (Tau-J) [inline mmpose/models/necks/fmap_proc_neck.py:9]: Please add comments and docstrings to the code in resize()

**Review by Tau-J — APPROVED** [review]
